### PR TITLE
Sema: Fix many-pointer array concatenation at comptime 

### PR DESCRIPTION
**Pull request by sin-ack**
This allows the following code to now compile properly:

```zig
export fn _start() void {
    _ = doTheTest("bar");
}

fn doTheTest(comptime s: [*:0]const u8) [*:0]const u8 {
        return "foo" ++ s;
}
```

### CODE REVIEW

**Review by Vexu — APPROVED** [review]
Could you add a behavior test for this?

File: src/Sema.zig [hunk 1]
```diff
@@ -9091,8 +9091,8 @@ fn zirArrayCat(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
             const rhs_len = try sema.usizeCast(block, lhs_src, rhs_info.len);
             const final_len = lhs_len + rhs_len;
             const final_len_including_sent = final_len + @boolToInt(res_sent != null);
-            const lhs_single_ptr = lhs_ty.zigTypeTag() == .Pointer and !lhs_ty.isSlice();
-            const rhs_single_ptr = rhs_ty.zigTypeTag() == .Pointer and !rhs_ty.isSlice();
+            const lhs_single_ptr = lhs_ty.zigTypeTag() == .Pointer and lhs_ty.isSinglePointer();
+            const rhs_single_ptr = rhs_ty.zigTypeTag() == .Pointer and rhs_ty.isSinglePointer();
```
Comment (andrewrk) [inline src/Sema.zig:9095]:
```suggestion
            const lhs_single_ptr = lhs_ty.isSinglePointer();
            const rhs_single_ptr = rhs_ty.isSinglePointer();
```

**Review by andrewrk — CHANGES_REQUESTED** [review]
Behavior test please!

**Comment by sin-ack** [comment]
Behavior test added.

**Review by andrewrk — APPROVED** [review]
Thanks for the behavior test!

**Pull request merged**